codex-cli/src/utils/config.ts

```diff
@@ -161,6 +161,7 @@ export type StoredConfig = {
   /** User-defined safe commands */
   safeCommands?: Array<string>;
   reasoningEffort?: ReasoningEffort;
+  editor?: "vscode" | "cursor" | "windsurf";
```

Comment:
Users would expect `$VISUAL` and failing that `$EDITOR` to be honored, I think.

If we can't support arbitrary editors in general because we want to open to a specific line, I think we need to clarify that.